### PR TITLE
Release Work for ADAL@3.1.3

### PR DESCRIPTION
**Pull request by iambmelt**
Sets `common` submodule pointer to `HEAD` of `release/3.4.5`

Still to do for this release:
- ~Update gradle version to match `common`~ c8498ab90d7fd61fd8f3633fa86e8163ba433386
- ~Pull in @shahzaibj's changes from https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1578 to unbreak `master` branch build with latest `common`~ d22ccf3b424ed3294609111cc604a4a3348338db
- ~Update gradle dependency to `common@3.4.5`~ 6bf5f520a2450a571253048cc8889cdb1bed1fc1
- ~Update ADAL version strings~ 54e49c7cee2d07d7616b6ff38f07662462a40f02
- ~Update ADAL changelog~ 586d0a09b29e607328b8a477b7c0c30eff4143ff
- ~Pull in changes from PR https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1606~
    * via 9b178cce4b8f74f5eeb5a8d446a122544438ca55
    * via 1de649a42c625525585765cb18fb5cb96c40013a
    * via fcbbdad62e081a40e5af00b80c6b75b5a7c308c1
    * via 5bfb209bf2230e9ca1ab2601cadb82e873c57e4d
- ~Pull in changes from PR https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1616~
    * via 9e0dc2bed2fec14b80553d333a1b4df51fe63645
    * via 04ef4808114cdad285177bba52ebfc50d317c8fc
    * via 517db8f9ec444fa0bfbb59cf05fc77c231f978ac
    * via 2f70ae3128bf0a5b2b645910a635c0e58704b29c
    * via 3b69fa73be9890480067949aedf3946877cf2aa3
    * via 57d4a532815aa4453b76cba34835482730841dc0
    * via b3ece828ef05700a5e6bf1916f51d83da14a25b8
    * via d28db750e91b879ed14a6e81753fadd6a95a9636
    * via 53ca97bc2d5477f1d4fb8924e57c154174bcd691

### CODE REVIEW

**Review by AdamBJohnsonx — APPROVED** [review]

**Comment by iambmelt** [comment]
@shahzaibj I had to adapt your changes from https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1578 to make this build work locally - if you could review...

File: gradle.properties [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 
-# When configured, Gradle will run in incubating parallel mode.
-# This option should only be used with decoupled projects. More details, visit
-# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
-# org.gradle.parallel=true
+# This the TSL versionName...
+versionName=1.5.9
+
+# For OneAuth default abiSelection
+abiSelection=x86_64
```
Comment (shahzaibj) [inline gradle.properties:18]: Should we have these changes checked in?

Comment (iambmelt) [reply gradle.properties:18]: These [are taken from `dev`](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/dev/gradle.properties) - [needed for desugaring support](https://github.com/AzureAD/azure-activedirectory-library-for-android/commit/426acaafb53e0c23c06882a4b8752403e08a379e)

**Review by shahzaibj — APPROVED** [review]

File: adal/src/androidTest/java/com/microsoft/aad/adal/TokenCacheAccessorTests.java [hunk 2]
```diff
@@ -128,6 +168,7 @@ static String createRawClientInfo(final String uid, final String utid) {
         MOCK_ID_TOKEN_WITH_CLAIMS = idTokenWithClaims;
     }
 
+    Context mContext;
```
Comment (aherciya) [inline adal/src/androidTest/java/com/microsoft/aad/adal/TokenCacheAccessorTests.java:171]: Is there a reason why mContext can't be final?

File: adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java [hunk 3]
```diff
@@ -513,7 +514,15 @@ private boolean checkIfBrokerHasLltChanges() {
         } catch (PackageManager.NameNotFoundException ignored) {
         }
 
-        return authVersionCode >= AUTHENTICATOR_LLT_VERSION_CODE && cpVersionCode >= CP_LLT_VERSION_CODE;
+        try {
+            PackageInfo brokerHostPackageInfo = packageManager.getPackageInfo(AuthenticationConstants.Broker.BROKER_HOST_APP_PACKAGE_NAME, 0);
+            brokerHostVersionCode = brokerHostPackageInfo.versionCode;
+        } catch (PackageManager.NameNotFoundException ignored) {
```
Comment (aherciya) [inline adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java:520]: Don't we at least want to log at the info level?

File: adal/src/androidTest/java/com/microsoft/aad/adal/BrokerProxyTests.java [hunk 4]
```diff
@@ -224,6 +225,7 @@ public void testCannotSwitchToBrokerWhenADFS()
     }
 
     @Test
+    @Ignore
```
[the remark this answers is not in the document]
Comment (iambmelt) [reply adal/src/androidTest/java/com/microsoft/aad/adal/BrokerProxyTests.java:228]: These are changes lifted verbatim from https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1578/commits/fcef4463daac3f3ae037710c324806eef7661bc0 - not looking to deviate from these changes as I'm just pulling them to unjam the build

File: adal/src/androidTest/java/com/microsoft/aad/adal/TokenCacheAccessorTests.java (hunk 2 above)
Comment (iambmelt) [reply adal/src/androidTest/java/com/microsoft/aad/adal/TokenCacheAccessorTests.java:171]: It is not initialized inside of a constructor (but rather, inside of the test class' `setUp()` function)

File: adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java (hunk 3 above)
Comment (iambmelt) [reply adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java:520]: This change lifted verbatim from @shahzaibj's changes in e42ed621a403f40b3659dd2c57cf0dcd4e26a229 to unjam the build - not looking to apply further changes as all I'm doing is pulling into this branch for deployment

**Comment by iambmelt** [comment]
@aherciya Please review that I have correctly merged your changes from https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1606

**Review by AdamBJohnsonx — APPROVED** [review]

**Review by aherciya — APPROVED** [review]

**Review by shoatman — APPROVED** [review]

**Pull request merged**